src/rust/lib/frp/src/nodes.rs

```diff
@@ -180,6 +180,15 @@ impl Network {
         self.all_with(label,t1,t2,|a,b| *a && *b)
     }
 
+    /// If the first input is false, the second input will be redirected to the output. Otherwise,
+    /// the third input will be redirected to the output. The redirection is persistent. The first
+    /// input doesn't have to fire to propagate the events fromm second and third input streams.
+    pub fn switch<T1,T2,T3,T>
+    (&self, label:Label, check:&T1, t2:&T2, t3:&T3) -> Stream<T>
+    where T1:EventOutput<Output=bool>, T2:EventOutput<Output=T>, T3:EventOutput<Output=T>, T:Data {
+        self.all_with3(label,check,t2,t3,|check,t1,t2| if *check {t2.clone()} else {t1.clone()})
+    }
+
```

Comment:
The docs gave me a little suggestion, that the `check` change won't emit any signal from `switch`, because _redirection_ should not add any signal to signals received from input.

But the implementation states quite contrary: change on `check` will emit (again) signal from t1 or t2.

Reply:
Good catch. Updated docs.